src/main/java/io/vlingo/http/resource/ServerActor.java

```diff
@@ -451,11 +451,18 @@ private boolean determineKeepAlive(final RequestResponseContext<?> requestRespon
 
     private ConsumerByteBuffer bufferFor(final Response response) {
       final int size = response.size();
-      if (size < maxMessageSize) {
-        return responseBufferPool.acquire("ServerActor#BasicCompletedBasedResponseCompletes#bufferFor");
+      try {
+        if (size < maxMessageSize) {
+          return responseBufferPool.acquire("ServerActor#BasicCompletedBasedResponseCompletes#bufferFor");
+        }
+        final int difference = size - maxMessageSize;
+        throw new IllegalStateException("Response exceeds maxMessageSize(" + maxMessageSize + ") by " + difference + ".");
+        
+      } catch (IllegalStateException e) {
```

Comment:
@Florian-Schoenherr Suggestion: use `Exception` rather than specific `IllegalStateException` to make sure you catch every possible failure.